service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -0,0 +1,178 @@
+import uk.gov.hmcts.reform.fpl.model.Judge;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.HearingBookingService;
+import uk.gov.hmcts.reform.fpl.service.ManageHearingsService;
+import uk.gov.hmcts.reform.fpl.service.StandardDirectionsService;
+import uk.gov.hmcts.reform.fpl.service.ValidateGroupService;
+import uk.gov.hmcts.reform.fpl.validation.groups.HearingDatesGroup;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static uk.gov.hmcts.reform.fpl.enums.HearingOptions.EDIT_DRAFT;
+import static uk.gov.hmcts.reform.fpl.service.HearingBookingService.HEARING_DETAILS_KEY;
+import static uk.gov.hmcts.reform.fpl.service.HearingBookingService.SELECTED_HEARING_IDS;
+import static uk.gov.hmcts.reform.fpl.service.ManageHearingsService.FIRST_HEARING_FLAG;
+import static uk.gov.hmcts.reform.fpl.service.ManageHearingsService.HEARING_DATE_LIST;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInGatekeepingState;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.getDynamicListValueCode;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper.buildAllocatedJudgeLabel;
+
+@Api
+@RestController
+@RequestMapping("/callback/manage-hearings")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManageHearingsController extends CallbackController {
+    private final ObjectMapper mapper;
+    private final ValidateGroupService validateGroupService;
+    private final HearingBookingService hearingBookingService;
+    private final StandardDirectionsService standardDirectionsService;
+    private final ManageHearingsService manageHearingsService;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (caseData.getAllocatedJudge() != null) {
+            caseDetails.getData().put("judgeAndLegalAdvisor", setAllocatedJudgeLabel(caseData.getAllocatedJudge()));
+        }
+
+        List<Element<HearingBooking>> hearings = defaultIfNull(caseData.getHearingDetails(), List.of());
+
+        if (hearings.isEmpty()) {
+            caseDetails.getData().put(FIRST_HEARING_FLAG, "Yes");
+        } else {
+            caseDetails.getData().putAll(manageHearingsService.populateInitialFields(caseData));
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/edit-hearing/mid-event")
+    public AboutToStartOrSubmitCallbackResponse populateExistingDraftHearing(
```

Comment:
If you want to shorten these lines so that the method declaration fits on one you can change the return type to just `CallbackResponse` which `AboutToStartOrSubmitCallbackResponse` implements